src/main/java/io/netty/buffer/api/Component.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.buffer.api;
+
+import java.nio.ByteBuffer;
+import java.util.function.Consumer;
+
+/**
+ * A view onto the buffer component being processed in a given iteration of {@link Buf#forEachReadable(Consumer)}.
+ * <p>
+ * Instances of this interface are allowed to be mutable behind the scenes, and the data is only guaranteed to be
+ * consistent within the given iteration.
+ */
+public interface Component {
```

Comment:
as this is used for read operations we may want to make it more clear that all the methods here return "read-only" things by adding it to the classname. Something like `ReadableComponent`, or `ReadOnlyComponent`

Reply:
Makes sense, because I've realised we also need a version for writable components.

Reply:
Thinking about this some more… We would probably like to have things like `IovArray` be able to abstract over these components, both for reading and writing. So a single component type that works for both is still desirable.

Reply:
I think we could do something similar to `Channel` which extends `WritableByteChannel` and `ReadableByteChannel`. So I would prefer to have a specific type for reading and one for writing.